webapp/components/form/AutocompleteDialog/AutocompleteDialog.js

```diff
@@ -0,0 +1,181 @@
+import './AutocompleteDialog.scss'
+
+import React, { useRef, useCallback, useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import { KeyboardMap } from '@webapp/utils/keyboardMap'
+import { clickedOutside, elementOffset } from '@webapp/utils/domUtils'
+
+const offsetByKey = {
+  [KeyboardMap.PageUp]: -10,
+  [KeyboardMap.PageDown]: 10,
+  [KeyboardMap.Up]: -1,
+  [KeyboardMap.Down]: 1,
+}
+
+const AutocompleteDialog = (props) => {
+  const {
+    inputField,
+    sourceElement,
+    items,
+    itemRenderer: ItemRenderer,
+    itemLabel,
+    itemKeyFunction,
+    className,
+    onItemSelect: selectItem,
+    onClose: close,
+  } = props
+  const list = useRef(null)
+  const [focusedItemIndex, setFocusedItemIndex] = useState(null)
+
+  const calculatePosition = useCallback(() => {
+    const { top, left, height, width } = elementOffset(sourceElement || inputField)
+    return {
+      top: top + height,
+      left,
+      width,
+    }
+  }, [sourceElement, inputField])
+
+  const focusItem = useCallback(
+    (index) => {
+      const itemEl = list.current && list.current.children && list.current.children[index]
```

Comment:
I suppose that itemEl should never be null... so calling list.current.children[index] should be safe.